src/main.rs

```diff
@@ -265,8 +267,14 @@ fn safe_main(program: &str, args: &[String]) -> Fallible<()> {
         return Err(UsageError { message: "invalid number of arguments".to_string() }.into());
     };
 
+    let profiler = match matches.opt_str("cpu_profile") {
```

Comment:
I was going to suggest .map here, but you can actually do this with a conditional assignment:

```rust
let profiler;
if let Some(path) = matches.opt_str("cpu_profile") {
  profiler = sandboxfs::ScopedProfiler::start(&path).context("Failed to start CPU profile")?;
}
```

Reply:
Done but...

I'm pretty confused about how this works as this makes it look as if `profiler` can be left uninitialized and then accessed. Is this only possible because `profiler` is never explicitly used?

Reply:
Correct.